include/ghex/structured/rma_put.hpp

```diff
@@ -39,8 +39,11 @@ using gpu_to_gpu = std::integral_constant<bool,
     std::is_same<typename SourceField::arch_type, gridtools::ghex::gpu>::value &&
     std::is_same<typename TargetField::arch_type, gridtools::ghex::gpu>::value>;
 
+// attributes needed for gcc to produce optimized code
 template<typename SourceField, typename TargetField>
-inline std::enable_if_t<
+__attribute__ ((optimize ("no-tree-loop-distribute-patterns")))
```

Comment:
should these be guarded with `__GNUC__` (I think this is the macro...)

Reply:
yes, of course. Will add